topdown/regex_template_test.go

```diff
@@ -0,0 +1,48 @@
+package topdown
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+)
+
+func TestRegexCompiler(t *testing.T) {
+	for _, tc := range []struct {
+		template       string
+		delimiterStart byte
+		delimiterEnd   byte
+		failCompile    bool
+		matchAgainst   string
+		failMatch      bool
+	}{
+		{"urn:foo:{.*}", '{', '}', false, "urn:foo:bar:baz", false},
+		{"urn:foo.bar.com:{.*}", '{', '}', false, "urn:foo.bar.com:bar:baz", false},
+		{"urn:foo.bar.com:{.*}", '{', '}', false, "urn:foo.com:bar:baz", true},
+		{"urn:foo.bar.com:{.*}", '{', '}', false, "foobar", true},
+		{"urn:foo.bar.com:{.{1,2}}", '{', '}', false, "urn:foo.bar.com:aa", false},
+		{"urn:foo.bar.com:{.*{}", '{', '}', true, "", true},
+		{"urn:foo:<.*>", '<', '>', false, "urn:foo:bar:baz", false},
+	} {
+		t.Run(fmt.Sprintf("template=%s", tc.template), func(t *testing.T) {
+			result, err := compileRegexTemplate(tc.template, tc.delimiterStart, tc.delimiterEnd)
+			if tc.failCompile != (err != nil) {
+				t.Logf("failed regex template compilation: %t != %t", tc.failCompile, err != nil)
+				t.FailNow()
```

Comment:
[nit] We could use [`t.Fatalf()`](https://golang.org/pkg/testing/#T.Fatalf), since
> Fatalf is equivalent to Logf followed by FailNow.